minio/api.py

```diff
@@ -1524,7 +1524,8 @@ def _stream_put_object(self, bucket_name, object_name,
 
             part_data = read_full(data, current_part_size)
             # Append current part information
-            parts_to_upload.append((bucket_name, object_name, upload_id, part_number, part_data))
+            parts_to_upload.append((bucket_name, object_name, upload_id,
+                                    part_number, part_data))
 
         # Run parts upload in parallel
         pool.parallel_run(self._upload_part_routine, parts_to_upload)
```

Comment:
pool.parallel_run() can raise an exception, I think it would be good to catch it, remove incomplete upload and throw the exception again.

```
try:
     pool.parallel_run(self._upload_part_routine, parts_to_upload)
except:
     self._remove_incomplete_upload(bucket_name,
                                           object_name,
                                           upload_id)
     raise
```

Reply:
done